modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.js

```diff
@@ -190,47 +96,103 @@ function Actions({actions, itemData, itemId, menuActive, onMenuActiveChange}) {
 		});
 	}
 
-	const handleClick = ({
-		action,
-		closeMenu,
-		event,
-		itemData,
-		itemId,
-		size = 'lg',
-	}) => {
-		if (onActionDropdownItemClick) {
-			onActionDropdownItemClick({
+	const handleClick = ({action, closeMenu, event}) => {
+		const {data, href, method, onClick, target} = action;
+
+		const {
+			confirmationMessage,
+			errorMessage,
+			status,
+			successMessage,
+			title,
+		} = data ?? {};
```

Comment:
Neat trick! I wonder how the rest of the code functions if `data` is falsy, do we need any sanity checks?

Reply:
There is one other place where `data` is used, and we do have sanity check:
 ```js
executeAsyncItemAction({
	...
	method: method ?? data?.method,
	...
});
```
This is one the things that can be simplified, there is no point in `method` being defined in two places in API. But, I'd put that outside of scope of this PR. We would need to check all usages, possibly change some of them.